pkgs/applications/science/misc/simgrid/default.nix

```diff
@@ -87,6 +86,7 @@ stdenv.mkDerivation rec {
   checkPhase = ''
     runHook preCheck
 
+    make tests -j $NIX_BUILD_CORES
```

Comment:
`enableParallelChecking` already does that and it's `true` by default.
So you just need to keep the default `checkPhase`.
You could move

```bash
ctest -j $NIX_BUILD_CORES --output-on-failure -E smpi-replay-multiple
```

to `postCheck`, remove the rest and it'll work.

Reply:
Unfortunately, the default `checkPhase` fails before calling postCheck.
It calls CMake tests without excluding `smpi-replay-multiple`, which fails.

Reply:
`--output-on-failure` is enabled by default in our nixpkgs toolchain.
To disable a test you can add `-DCTEST_CUSTOM_TESTS_IGNORE=smpi-replay-multiple` to the `cmakeFlags`

Reply:
Took me a while but I solved it using those two resources:
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/cmake/setup-hook.sh
- https://gitlab.kitware.com/cmake/community/wikis/doc/ctest/Testing-With-CTest

Reply:
I tried ([patch](https://pastebin.com/qSJaTT90)) to set `CTEST_CUSTOM_TESTS_IGNORE` (and friends: `CTEST_CUSTOM_MEMCHECK_IGNORE`, `CTEST_CUSTOM_COVERAGE_EXCLUDE`) via `cmakeFlags` but this is ignored by the default `checkPhase`: It runs the test and fails.

I could generate a `CTestCustom.cmake` in the build tree but I'm not convinced this is the way to go. I feel like it would add more complexity than redefining `checkPhase`.

Reply:
Yeah okay, sorry. It doesn't work :/

But it doesn't have be be an entirely new `cmake` file, right? Can't you just append a single line to the current one?
If not just leave it as it is right now.

Reply:
Should be fixed by 00fc5bd08fc, which ignores failing tests by overriding `preCheck` instead of `checkPhase`. `preCheck` defines `CTEST_CUSTOM_TESTS_IGNORE` appropriately in `CTestCustom.cmake`.